class/practice_6.1/include/message_dispatcher.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <iostream>
+#include <thread>
+#include <random>
+#include <functional>
+#include <vector>
+#include <mutex>
+
+struct GreenMessage {
+    std::string text;
+    int counter;
+};
+
+struct BlueMessage {
+    double value1;
+    double value2;
+};
+
+struct OrangeMessage {
+    std::string text1;
+    std::string text2;
+    int integer;
+    double value;
+};
+
+class MessageDispatcher {
+public:
+    template <typename T>
+    void subscribeGreen(T callback) {
+        std::lock_guard<std::mutex> lock(greenMutex);
+        greenSubscribers.push_back(callback);
+    }
+
+    template <typename T>
+    void subscribeBlue(T callback) {
+        std::lock_guard<std::mutex> lock(blueMutex);
+        blueSubscribers.push_back(callback);
+    }
+
+    template <typename T>
+    void subscribeOrange(T callback) {
+        std::lock_guard<std::mutex> lock(orangeMutex);
+        orangeSubscribers.push_back(callback);
+    }
+
+    void publishGreen(const GreenMessage& message) {
+        std::lock_guard<std::mutex> lock(greenMutex);
+        for (auto& subscriber : greenSubscribers) {
```

Comment:
It may be more reasonable to send message in other thread to make publish an instant operation cause some callbacks may take a long tiime and hang this thread.